src/main/java/net/dv8tion/jda/core/entities/MessageActivity.java

```diff
@@ -0,0 +1,246 @@
+/*
+ *     Copyright 2015-2018 Austin Keener & Michael Ritter & Florian Spieß
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.dv8tion.jda.core.entities;
+
+import net.dv8tion.jda.core.utils.Checks;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents a {@link net.dv8tion.jda.core.entities.Message} activity.
+ */
+public class MessageActivity
+{
+    private final ActivityType type;
+    private final String partyId;
+    private final Application application;
+
+    /**
+     *
+     * @param  type the {@link net.dv8tion.jda.core.entities.MessageActivity.ActivityType}
+     *         for the {@link MessageActivity}.
+     *
+     * @param  partyId the partyId for the {@link MessageActivity}.
+     *
+     * @param  application the {@link net.dv8tion.jda.core.entities.MessageActivity.Application}
+     *         for the {@link MessageActivity}.
+     *
+     * @throws IllegalArgumentException
+     *         <ul>
+     *             <li>If {@link net.dv8tion.jda.core.entities.MessageActivity.ActivityType ActivityType} requires a not null {@link net.dv8tion.jda.core.entities.MessageActivity.Application Application}.</li>
+     *             <li>If {@link net.dv8tion.jda.core.entities.MessageActivity.ActivityType ActivityType} gets a not required not null {@link net.dv8tion.jda.core.entities.MessageActivity.Application Application}.</li>
+     *         </ul>
+     */
+    MessageActivity(ActivityType type, String partyId, Application application)
```

Comment:
There is no need to document package-private constructors

Reply:
Yeah whups my bad. Should be resolved now.